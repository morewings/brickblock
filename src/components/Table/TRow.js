import React from 'react';
import PropTypes from 'prop-types';

const TRow = props => (
  <tr>
    <td>{props.address}</td>
    <td>{props.currency}</td>
    <td>{props.value}</td>
    <td>{props.txid}</td>
  </tr>
);

TRow.propTypes = {
  address: PropTypes.string,
  currency: PropTypes.string,
  value: PropTypes.number,
  txid: PropTypes.string,
};

TRow.defaultProps = {
  address: '',
  currency: '',
  value: NaN,
  txid: '',
};

export default TRow;
