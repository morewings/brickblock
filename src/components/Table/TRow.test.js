import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import TRow from 'components/Table/TRow';

describe('components > Table > TRow', () => {
  const defaultProps = {
    address: 'address',
    currency: 'currency',
    value: 9000,
    txid: 'txid',
  };
  it('renders without crashing', () => {
    const wrapper = shallow(<TRow {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
