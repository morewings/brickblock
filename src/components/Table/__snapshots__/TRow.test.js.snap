// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Table > TRow renders without crashing 1`] = `
<tr>
  <td>
    address
  </td>
  <td>
    currency
  </td>
  <td>
    9000
  </td>
  <td>
    txid
  </td>
</tr>
`;
